scripting/get5_apistats.sp

```diff
@@ -138,11 +141,18 @@ static Handle CreateRequest(EHTTPMethod httpMethod, const char[] apiMethod, any:
 
 int RequestCallback(Handle request, bool failure, bool requestSuccessful,
                     EHTTPStatusCode statusCode) {
-  if (failure || !requestSuccessful) {
-    LogError("API request failed, HTTP status code = %d", statusCode);
-    char response[1024];
-    SteamWorks_GetHTTPResponseBodyData(request, response, sizeof(response));
-    LogError(response);
+  int status = view_as<int>(statusCode);
+  if (failure || !requestSuccessful || status >= 300 || status < 200) {
+    LogError("Steamworks GET request failed with HTTP status code: %d.", statusCode);
+    int responseSize;
+    SteamWorks_GetHTTPResponseBodySize(request, responseSize);
+    char[] response = new char[responseSize];
+    if (SteamWorks_GetHTTPResponseBodyData(request, response, responseSize)) {
+      LogError("Response body: %s", response);
+    } else {
+      LogError("Failed to read response body.");
+    }
+    delete request;
```

Comment:
Unnecessary `return`, and you should delete the request also if it has failed, so it must go outside your error check.